app/models/background.rb

```diff
@@ -1,6 +1,6 @@
 class Background < ApplicationRecord
-  has_and_belongs_to_many :samples, through: :pipeline_runs
   has_and_belongs_to_many :pipeline_runs
+  has_many :samples, through: :pipeline_runs
```

Comment:
I'm not sure this is right. My understanding is that has_many means a one-to-many relationship. This is not correct here because it is actually a many-to-many relationship. To me this change seems conceptually wrong and I think it is too dangerous to make, especially since there are no actual bugs in the product that derive from it.

Reply:
@cdebourcy https://flatironschool.com/blog/why-you-dont-need-has-and-belongs-to-many

We actually should completely move away from `has_and_belongs_to_many` since it creates a weird unordered relation. The logic here is the same but through an explicit table that can provide way more information. 
I had this problem as I was trying to understand project_users relationships.

Reply:
As @gregdingle said `has_and_belongs_to_many` does not have a through field.

Reply:
Ok. Just make sure it's safe. I agree we should not take failing tests lightly